broadleaf-thymeleaf3-presentation/src/main/java/org/broadleafcommerce/presentation/thymeleaf3/resolver/BroadleafClassLoaderTemplateResource.java

```diff
@@ -0,0 +1,115 @@
+package org.broadleafcommerce.presentation.thymeleaf3.resolver;
+
+import org.broadleafcommerce.common.util.EfficientLRUMap;
+import org.thymeleaf.templateresource.ClassLoaderTemplateResource;
+import org.thymeleaf.templateresource.ITemplateResource;
+import org.thymeleaf.util.ClassLoaderUtils;
+import org.thymeleaf.util.StringUtils;
+import org.thymeleaf.util.Validate;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.Map;
+
+public class BroadleafClassLoaderTemplateResource implements ITemplateResource {
```

Comment:
this all looks like copy-paste from ClassLoaderTemplateResource, so for better support of upgrading thymeleaf to future versions I would think about delegate.
Just create an instance of ClassLoaderTemplateResource and delegate every method call to it, and override only exists method.
Also having @override annotation could be handy to understand what method is from interface and what methods are added